src/controller/CHIPDeviceController.cpp

```diff
@@ -1557,6 +1557,41 @@ void DeviceCommissioner::SendCommissioningCompleteCallbacks(NodeId nodeId, const
     }
 }
 
+void DeviceCommissioner::PauseCommissioning()
+{
+    VerifyOrReturn(mDeviceBeingCommissioned != nullptr);
+    mCommissioningPaused = true;
```

Comment:
I exposed the FailSafe timer in android's ControllerParams.java so that the app can set this to higher values (like 5 mins). If an app needs more time than that, then yes, it can always increase the timer from a delegate. So far we haven't encountered that situation.